src/Network/X509/SelfSigned.hs

```diff
@@ -830,3 +855,11 @@ validateX509CertChainPem (X509CertChainPem a b) =
     case traverse_ decodePemX509Cert (a : b) of
         Left e -> throwError $ sshow e
         Right () -> return ()
+
+-- -------------------------------------------------------------------------- --
+-- Utils
+
+trim :: B.ByteString -> B.ByteString
```

Comment:
Unit test please (unless this is already covered in Quickcheck somehow)

Reply:
`isSpace` is slow, for ascii prefer the following from attoparsec:

```haskell
isSpace_w8 :: Word8 -> Bool
isSpace_w8 w = w == 32 || w - 9 <= 4
{-# INLINE isSpace_w8 #-}
```

If you want this function to be really fast:

```haskell
trim :: B.ByteString -> B.ByteString
trim s = t
  where
    predicate = not . isSpace_w8
    l = B.length s
    b = fromMaybe l $ B.findIndex predicate s
    -- mysteriously bytestring has `findIndex` but not `findIndexEnd`
    e = maybe 0 (+1) $ findIndexEnd predicate s
    !t = if e <= b then B.empty else B.unsafeTake (e - b) $! B.unsafeDrop b s


findIndexEnd :: (Word8 -> Bool) -> ByteString -> Maybe Int
findIndexEnd k (PS x s l) =
    accursedUnutterablePerformIO $ withForeignPtr x $ \f ->
    let beg = f `plusPtr` s
        e   = beg `plusPtr` (l - 1)
        go :: Ptr Word8 -> IO (Maybe Int)
        go p = if p < beg then return Nothing
               else do
                   w <- peek p
                   if k w
                     then return $! Just $! p `minusPtr` beg
                     else go (p `plusPtr` (-1))
    in go e
{-# INLINE findIndexEnd #-}
```

Reply:
I am not sure if this is worth it. PEM is a textual encoding that we only use for passing around configurations. This happens in the context of reading certificates from file or form the command line, which is slow in any case and where we don't care about perf. 

All more serious computations with X509 data is done either in DER encoding, or, actually, after parsing the ASN.1 data into proper data structures.

The motivation for this PR was only for checking roundtrip consistency of data structure that happens to contain PEM data. So we needed this to make those test pass. But we care more about the other contents of those structures.

Reply:
I used quick check to test the implementation of `trim` and added a haddock property that documents the test.